src/property.h

```diff
@@ -141,6 +142,63 @@ class FntProperties : public Properties {
   public:
 };
 
+/** Iterator on Properties
+ * @note it assumes that items are stored in an increasing order of their keys.
+ * For reading, items are retrieved in the same order. Each item is retrieved
+ * once.
+ */
+class PropsIterator {
+  public:
+    PropsIterator(const Properties& props)
+    {
+        index = 0;
+        m_key = 0;
+        m_val = 0;
+
+        keys = props.get_keys();
+        values = props.get_values();
+        items_nb = keys.size();
+
+        assert(keys.size() == values.size());
+    }
+
+    inline const size_t& curr_loc() const
+    {
+        return (index < items_nb) ? keys.at(index) : m_key;
```

Comment:
uses `[]` instead of `at` since you have already done the bound-check manually.

Same remarks for `curr_mark` and `is_marked`.

Also, you can only get the current, so just rename the method `location` and `curr_mark` to `marker`.

Reply:
got it, thanks

- [ ] use `[]`